src/backtrace/unwind.rs

```diff
@@ -87,21 +95,17 @@ pub fn target(core: &mut Core, elf: &Elf, active_ram_region: &Option<RamRegion>)
 
         log::debug!("LR={lr:#010X} PC={pc:#010X}");
 
-        if lr == registers::LR_END {
-            break;
-        }
-
         // Link Register contains an EXC_RETURN value. This deliberately also includes
         // invalid combinations of final bits 0-4 to prevent futile backtrace re-generation attempts
         let exception_entry = lr >= cortexm::EXC_RETURN_MARKER;
 
         let program_counter_changed = !cortexm::subroutine_eq(lr, pc);
 
-        // If the frame didn't move, and the program counter didn't change, bail out (otherwise we
-        // might print the same frame over and over).
-        output.corrupted = !cfa_changed && !program_counter_changed;
-
-        if output.corrupted {
+        // If the frame didn't move, and the program counter didn't change, bail out
+        // (otherwise we might print the same frame over and over).
+        if !cfa_changed && !program_counter_changed {
+            // If we do not end up in the reset function the stack is corrupted
+            output.corrupted = !reset_range.contains(&pc);
```

Comment:
this is not due to this PR but I wouldn't necessarily categorize this exit condition as "corrupted". if you compile the rust program with `-C force-frame-pointers=off` then it's not possible to unwind the stack because frame pointer information is missing (from register r7) so you'll hit this branch but the call stack and the program will both operate fine.

`-C force-frame-pointers=off` is not very common but the assembly trampolines used to implement context switches in RTOSes / kernels could have the same effect of making unwinding not work.

I think it would be more to correct to simply say in this branch that it's "not possible to further unwind the stack" or something like that

Reply:
I opened https://github.com/knurling-rs/probe-run/issues/387 for this.